packages/flutter_tools/test/integration.shard/web_plugin_registrant_test.dart

```diff
@@ -255,32 +306,76 @@ Future<void> _createProject(Directory dir, List<String> createArgs) async {
   ]);
 }
 
-Future<void> _addDependency(
+typedef PubspecEditor = void Function(List<String> pubSpecContents);
+
+Future<void> _editPubspecFile(
   Directory projectDir,
-  String package, {
-  String? version,
-  String? path,
-}) async {
+  PubspecEditor editor,
+) async {
+  final File pubspecYaml = projectDir.childFile('pubspec.yaml');
+  expect(pubspecYaml, exists);
+
+  final List<String> lines = await pubspecYaml.readAsLines();
+  editor(lines);
+  await pubspecYaml.writeAsString(lines.join('\n'));
+}
+
+Future<void> _replaceMainFile(Directory projectDir, String fileContents) async {
+  final File mainFile = projectDir.childDirectory('lib').childFile('main.dart');
+  await mainFile.writeAsString(fileContents);
+}
+
+PubspecEditor _addDependencyEditor(String packageToAdd, {String? version, String? path}) {
   assert(version != null || path != null,
       'Need to define a source for the package.');
   assert(version == null || path == null,
       'Cannot only load a package from path or from Pub, not both.');
+  void editor(List<String> lines) {
+    for (int i = 0; i < lines.length; i++) {
+      final String line = lines[i];
+      if (line.startsWith('dependencies:')) {
+        lines.insert(
+            i + 1,
+            '  $packageToAdd: ${version ?? '\n'
+                '   path: $path'}');
+        break;
+      }
+    }
+  }
+  return editor;
+}
 
-  final File pubspecYaml = projectDir.childFile('pubspec.yaml');
-  expect(pubspecYaml, exists);
+PubspecEditor _setDartSDKVersionEditor(String version) {
+  void editor(List<String> lines) {
+    for (int i = 0; i < lines.length; i++) {
+      final String line = lines[i];
+      if (line.startsWith('environment:')) {
+        for (i++; i < lines.length; i++) {
+          final String innerLine = lines[i];
+          final String sdkLine = "  sdk: '$version'";
+          if(innerLine.isNotEmpty && !innerLine.startsWith('  ')) {
+            lines.insert(i, sdkLine);
+            break;
+          }
+          if(innerLine.startsWith('  sdk:')) {
+            lines[i] = sdkLine;
+            break;
+          }
+        }
+        break;
+      }
+    }
+  }
+  return editor;
+}
 
-  final List<String> lines = await pubspecYaml.readAsLines();
-  for (int i = 0; i < lines.length; i++) {
-    final String line = lines[i];
-    if (line.startsWith('dependencies:')) {
-      lines.insert(
-          i + 1,
-          '  $package: ${version ?? '\n'
-              '    path: $path'}');
-      break;
+PubspecEditor _composeEditors(Iterable<PubspecEditor> editors) {
+  void editor(List<String> lines) {
+    for (final PubspecEditor e in editors) {
```

Comment:
nit: avoid abbreviations: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abbreviations